### PR TITLE
Scripts/SmartAI: SMART_TARGET_INVOKER

### PR DESCRIPTION
**Pull request by IntelligentQuantum**
## Changes Proposed:
-  
-  

## Issues Addressed:

- Closes 

## SOURCE:

https://github.com/TrinityCore/TrinityCore/commit/482328bddf1c5fcd367268525d2f904793239bc9
## Tests Performed:

- 
- 

## How to Test the Changes:

1.
2.
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Review by Nyeriah — APPROVED** [review]

**Comment by Nyeriah** [comment]
Gotta fix those start up errors and ensure those actions are working still :)

File: src/server/game/AI/SmartScripts/SmartAI.cpp [hunk 1]
```diff
@@ -1049,7 +1049,8 @@ void SmartAI::StopFollow(bool complete)
     if (!complete)
         return;
 
-    if (Player* player = ObjectAccessor::GetPlayer(*me, mFollowGuid))
```
Comment (claudiodfc) [inline src/server/game/AI/SmartScripts/SmartAI.cpp:1052]: Is this change necessary? If player is nullptr it's always false

**Comment by IntelligentQuantum** [comment]
@Nyeriah done

**Comment by pklloveyou** [comment]
SQL destroys a lot of things

**Comment by pklloveyou** [comment]
https://github.com/azerothcore/azerothcore-wotlk/pull/10038 It should also be rechecked
